test/unit/test_put_get.py

```diff
@@ -74,3 +78,16 @@ def test_put_error(tmpdir):
         sf_file_transfer_agent.result()
 
     chmod(file1, 0o700)
+
+
+def test_percentage(tmpdir):
+    """Tests for raise_put_get_error flag (now turned on by default) in SnowflakeFileTransferAgent."""
+    tmp_dir = str(tmpdir.mkdir('putfiledir'))
+    file_name = path.join(tmp_dir, 'zero_file1')
+    from pathlib import Path
+    Path(file_name).touch()
+    func_callback = SnowflakeS3ProgressPercentage(file_name, 0)
+    func_callback(1)
+
+    func_callback = SnowflakeAzureProgressPercentage(file_name, 0)
+    func_callback(1)
```

Comment:
Check out [tmp_path fixture](https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture), it'll cut out a bunch of your code. Also please be aware of our existing type hints. The percentage classes expect a `str` path.
```suggestion
def test_percentage(tmp_path):
    """Tests for raise_put_get_error flag (now turned on by default) in SnowflakeFileTransferAgent."""
    file = tmp_path / 'zero_file1'
    file.touch()
    func_callback = SnowflakeS3ProgressPercentage(str(file), 0)
    func_callback(1)

    func_callback = SnowflakeAzureProgressPercentage(str(file), 0)
    func_callback(1)
```